### DISCUSSION TITLE
Embed GitHub markdown doc within a page

### DISCUSSION THREAD

**Discussion by dgurudot** [requarks/wiki#7360] [Help / Questions] [answered]
I am trying to embed the Github markdown content for documentation within a page.
However, it isn't working via HTML frame or javascript for some reason. Kindly help and let me know what I might be doing wrong and I looked around for solutions.

I am sharing a sample code for example:

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Embed GitHub Markdown</title>
    <script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
    <style>
        body {
            font-family: Arial, sans-serif;
            padding: 20px;
        }
        #markdown-content {
            border: 1px solid #ddd;
            padding: 10px;
            border-radius: 5px;
            background-color: #f9f9f9;
        }
    </style>
</head>
<body>

<h1>GitHub Markdown Content</h1>
<div id="markdown-content">Loading...</div>

<script>
    // Define the new GitHub raw markdown URL and the proxy URL
    const markdownURL = 'https://raw.githubusercontent.com/requarks/wiki-docs/refs/heads/master/home.md';
    const proxyURL = 'https://corsproxy.io/?';

    // Log the final URL for debugging
    console.log('Fetching Markdown content from:', proxyURL + encodeURIComponent(markdownURL));

    // Fetch and render the markdown content
    fetch(proxyURL + encodeURIComponent(markdownURL))
        .then(response => {
            // Check if the response is ok
            if (!response.ok) {
                throw new Error('Network response was not ok: ' + response.status + ' - ' + response.statusText);
            }
            return response.text();
        })
        .then(mdContent => {
            // Parse and render the markdown content
            document.getElementById('markdown-content').innerHTML = marked(mdContent);
        })
        .catch(error => {
            // Display an error message if something goes wrong
            document.getElementById('markdown-content').innerHTML = 'Error loading Markdown content';
            console.error('Error fetching the content:', error);
        });
</script>

</body>
</html>

**Comment by dgurudot** [comment]
Any comments? It's a simple request to re-use the markdown content from GITHub without doing the sync. Please help.

**Comment by NGPixel** [comment, marked as answer]
You can't run javascript inside a page. Use the script feature from the Page properties dialog instead. 

However what you're trying to do is very hacky and a terrible way to include content. Just link to the page instead...
